### PR TITLE
feat(examples): add template collection for common sandbox scenarios (#645)

### LINKED ISSUE #645: 【2026犀牛鸟issue】CubeSandbox 沙箱模板与示例生态
【产品名称/功能模块】
CubeSandbox 沙箱模板与示例生态

【问题描述】
一个沙箱平台的易用性，很大程度上取决于其开箱即用的模板与示例是否丰富——用户能否快速找到一个贴合自身场景的模板（某种语言运行时、数据科学环境、浏览器自动化、代码解释器、Web 开发环境、特定 Agent 框架的运行底座等）直接拉起沙箱，决定了上手成本。CubeSandbox 已具备从 OCI 镜像构建模板、快照、恢复等能力，并已有少量示例（如 code-sandbox-quickstart、browser-sandbox、mini-rl-training、openai-agents-example 等），但覆盖的场景与语言栈仍然有限，尚未形成系统化的模板/示例生态。

本 issue 为开放性生态贡献任务，目标是持续丰富 CubeSandbox 的模板与示例库，覆盖更多语言、框架与典型场景，让不同背景的开发者都能找到可直接复用的起点。允许多人并行参与、各自认领一个模板或示例，以增量方式合入，不要求单次提交完整 PR。

【期望内容/功能改进】（每人可认领其中一个或多个方向，渐进式贡献）
1）贡献一个或多个开箱即用的沙箱模板，覆盖但不限于：常见语言运行时（Python/Node/Go/Rust/Java 等）、数据科学与 ML（Jupyter、PyTorch、pandas 等）、浏览器自动化、代码解释器、Web 开发环境、特定 Agent 框架的运行底座等；
2）每个模板/示例应包含：模板定义与构建方式、最小可运行调用流程、README 说明（适用场景、资源建议、已知限制），并能在本地部署环境跑通；
3）鼓励覆盖差异化场景：长时任务与快照断点续跑、有状态工作区、多容器协作、出口网络策略受限下的运行等，体现 CubeSandbox 的差异化能力；
4）模板应遵循平台既有模板构建与命名约定，与出口管控、鉴权、资源限制等机制对齐，避免安全策略缺口；
5）在 docs/guide/tutorials 或 examples 体系中登记，保持风格一致、便于检索；本方向聚焦通用语言栈/场景模板，与"Claude Code / CodeBuddy / OpenCode 集成"方向互补、不重复。

【参与说明与验收】
1）本题为开放参与：多人可同时认领不同语言/框架/场景，贡献以增量合入，不要求单人完成全部内容；
2）每份贡献应做到：模板可成功构建并拉起沙箱、示例可复现、文档与当前版本行为一致；
3）单份贡献的最低门槛：至少一个可构建的模板 + 可运行示例 + README，其他同学可依照文档在合理时间内跑通；
4）建议在 issue 下认领具体方向，避免与已有示例重复或撞车。

【认领规则】 
本issue为2026犀牛鸟开源人才培养活动专属issue，仅供已报名参与犀牛鸟活动的同学认领
【认领时间】7月1日～7月31日（7月1日前认领视为无效❗）
【认领方式】在本issue评论区回复“已认领本任务”，即视为认领成功
【活动报名】需提前完成犀牛鸟报名问卷，问卷将用于活动登记和奖励发放：https://wj.qq.com/s2/26888567/gh2q
【微信添加Cube社区小助手进入交流群】：CubeSandbox

---

**【Product Name / Feature Module】**
CubeSandbox Sandbox Templates & Example Ecosystem

**【Problem Description】**
A sandbox platform's usability depends heavily on how rich its out-of-the-box templates and examples are—whether users can quickly find a template that fits their scenario (a language runtime, data science environment, browser automation, code interpreter, web dev environment, runtime base for a specific Agent framework, etc.) and spin up a sandbox directly determines onboarding cost. CubeSandbox already supports building templates from OCI images, snapshots, restore, and similar capabilities, and has a few examples (e.g., code-sandbox-quickstart, browser-sandbox, mini-rl-training, openai-agents-example), but coverage of scenarios and language stacks is still limited and has not yet formed a systematic template/example ecosystem.

This issue is an open ecosystem contribution task aimed at continuously enriching CubeSandbox's template and example library, covering more languages, frameworks, and typical scenarios so developers from different backgrounds can find reusable starting points. Multiple contributors may participate in parallel, each claiming a template or example and merging incrementally—a single PR covering everything is not required.

**【Expected Content / Feature Improvements】** (Each contributor may claim one or more directions and contribute incrementally)
1) Contribute one or more out-of-the-box sandbox templates, covering but not limited to: common language runtimes (Python/Node/Go/Rust/Java, etc.), data science and ML (Jupyter, PyTorch, pandas, etc.), browser automation, code interpreters, web development environments, runtime bases for specific Agent frameworks, etc.;
2) Each template/example should include: template definition and build instructions, a minimal runnable workflow, README (use cases, resource recommendations, known limitations), and must work in a local deployment;
3) Encourage differentiated scenarios: long-running tasks with snapshot checkpoint/resume, stateful workspaces, multi-container collaboration, running under restricted egress policies, etc., showcasing CubeSandbox's differentiated capabilities;
4) Templates should follow the platform's existing template build and naming conventions, align with egress control, authentication, resource limits, and other mechanisms, and avoid security policy gaps;
5) Register in the `docs/guide/tutorials` or `examples` system with consistent style and easy discovery; this direction focuses on general language stacks/scenario templates, complementary to and non-overlapping with the "Claude Code / CodeBuddy / OpenCode Integration" direction.

**【Participation Guidelines & Acceptance】**
1) This is an open contribution: multiple people may claim different languages/frameworks/scenarios concurrently; contributions merge incrementally— no single person is expected to complete everything;
2) Each contribution should: build successfully and launch sandboxes, be reproducible, and have documentation consistent with current version behavior;
3) Minimum bar per contribution: at least one buildable template + runnable example + README, so others can follow the docs and get it working within a reasonable time;
4) Claim your specific direction in the issue comments to avoid duplicating existing examples or overlapping with others.

**【Claim Rules】**
This issue is exclusive to the 2026 Tencent Rhino Bird Open Source Talent Training Program and may only be claimed by students who have registered for the Rhino Bird program.
【Claim Period】July 1 – July 31 (claims before July 1 are invalid ❗)
【How to Claim】Reply "已认领本任务" (I have claimed this task) in the comments on this issue to claim successfully
【Program Registration】Complete the Rhino Bird registration questionnaire in advance; it will be used for program enrollment and reward distribution: https://wj.qq.com/s2/26888567/gh2q
【Join the Cube Community Group on WeChat】Add the Cube community assistant on WeChat (WeChat ID: CubeSandbox) to join the discussion group

### PR DESCRIPTION
**Pull request by xyaohubery**
## Summary
Add Dockerfile templates for common sandbox scenarios to improve onboarding.

## Templates
- `python-data-science`: pandas, numpy, matplotlib, scikit-learn, seaborn, plotly
- `nodejs-server`: Node.js 22 + npm, yarn, pnpm
- `go-dev`: Go development environment

## Usage
```bash
docker build -t my-registry/sandbox-python:latest python-data-science/
docker push my-registry/sandbox-python:latest
cubemastercli tpl create-from-image --image my-registry/sandbox-python:latest ...
```

Fixes #645

Assisted-by: claude-code:deepseek-v4-pro

### CODE REVIEW

**Comment by fslongjin** [comment]
Hello, thank you for your contribution! Since we have a large number of participants in this event, to speed up the PR review process, could you please attach some screenshots and logs (both are needed) showing the verification process and results for this PR? Thank you for your participation!

**Comment by xyaohubery** [comment]
Hi @fslongjin, sorry for the late reply. Same as my other PRs — I'm on a Windows machine without access to a Linux/KVM environment, so I'm unable to run a real CubeSandbox cluster to capture the verification screenshots and logs you requested. CI is green and the auto-review bot feedback has been addressed in follow-up commits. Would it be possible to proceed with CI + code review, or could someone on the team help verify on a real cluster? Happy to make any code changes needed. Thanks!

**Comment by xyaohubery** [comment]
@fslongjin I'd like to request reopening this PR.

The changes are purely additive — three Dockerfiles with documentation under . They do not touch any core code paths. The Dockerfiles follow project best practices (GPG-pinned APT repos, pinned versions, apt cleanup) as confirmed by the auto-review bot.

CI is green and all auto-review bot feedback was addressed in follow-up commits. I understand verification screenshots were requested, but these are standard Dockerfiles that can be verified via `docker build` without a full CubeSandbox cluster. I'm happy to:
1. Provide `docker build` logs if that would satisfy the verification requirement
2. Make any code changes the reviewers identify

Could you please reconsider, or let me know specifically what verification is needed for Dockerfile-only changes?

Co-Authored-By: Claude <noreply@anthropic.com>
